extensions/AzureAISearch/AzureAISearch/AzureAISearchMemory.cs

```diff
@@ -144,6 +144,33 @@ public async Task<string> UpsertAsync(string index, MemoryRecord record, Cancell
         return record.Id;
     }
 
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> BatchUpsertAsync(
```

Comment:
nit: refactor `UpsertAsync` (the method above) to call this one, removing code duplication/reusing the same logic